tuf/api/metadata.py

```diff
@@ -1289,32 +1291,59 @@ class DelegatedRole(Role):
 
     def __init__(
         self,
-        name: str,
         keyids: List[str],
         threshold: int,
         terminating: bool,
+        name: Optional[str] = None,
         paths: Optional[List[str]] = None,
         path_hash_prefixes: Optional[List[str]] = None,
+        succinct_hash_info: Optional[Dict[str, Any]] = None,
         unrecognized_fields: Optional[Mapping[str, Any]] = None,
     ):
         super().__init__(keyids, threshold, unrecognized_fields)
+        if name is None and succinct_hash_info is None:
+            raise ValueError(
+                "Name must be set when succinct_hash_info is not used"
+            )
         self.name = name
         self.terminating = terminating
         if paths is not None and path_hash_prefixes is not None:
             raise ValueError("Either paths or path_hash_prefixes can be set")
+        if paths is not None and succinct_hash_info is not None:
+            raise ValueError("Either paths or succinct_hash_info can be set")
+        if path_hash_prefixes is not None and succinct_hash_info is not None:
+            raise ValueError(
+                "Either path_hash_prefixes or succinct_hash_info allowed"
+            )
 
-        if paths is None and path_hash_prefixes is None:
-            raise ValueError("One of paths or path_hash_prefixes must be set")
+        if (
+            paths is None
+            and path_hash_prefixes is None
+            and succinct_hash_info is None
+        ):
+            raise ValueError(
+                "paths, path_hash_prefixes or succinct_hash_info required"
+            )
```

Comment:
if you want to remove some code, something like this could work:
```python
exclusive_vars = [paths, path_hash_prefixes, succinct_hash_info]
if sum(1 for var in exclusive_vars if var is not None) != 1:
    raise ValueError("One of (paths, path_hash_prefixes, succinct_hash_info) must be set")
```

Reply:
Loved this simplification.
Great suggestion!